### PR TITLE
Add electron example

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
This contains a todo list example for using Electron.  

DRAFT TODOS:
* [ ] Fix `useQuery` re-rendering issue
* [ ] Attempt to use Webpack and implement hot reloading

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: examples/electron-todo-list/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Electron Example
+
+This Electron examples shows how to use Device Sync and `@ream/react` in an Electron application.
```
Comment (kneth) [inline examples/electron-todo-list/README.md:3]:
```suggestion
This Electron example shows how to use Device Sync and `@ream/react` in an Electron application.
```

File: examples/electron-todo-list/src/renderer/atlas-app-services/config.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "ATLAS_APP_ID": "example-wasm-nwvek"
```
Comment (kneth) [inline examples/electron-todo-list/src/renderer/atlas-app-services/config.json:2]: Should we replace with `YOUR-APP-ID-HERE`?

File: examples/electron-todo-list/src/renderer/components/AddTaskForm.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <form className={styles.form} onSubmit={handleSubmit}>
+      <input
+        className={styles.input}
+        type='text'
+        placeholder='Add a new task'
+        value={description}
+        onChange={(event) => setDescription(event.currentTarget.value)}
+        autoCorrect='off'     // Safari only
```
Comment (kneth) [inline examples/electron-todo-list/src/renderer/components/AddTaskForm.tsx:44]: Is this needed?

File: examples/electron-todo-list/src/renderer/components/IntroText.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+//
+////////////////////////////////////////////////////////////////////////////
+
+import styles from '../styles/IntroText.module.css';
+
+export function IntroText() {
+  return (
+    <div className={styles.container}>
+      <p>
+        Welcome to a MongoDB Realm, WASM, and Sync app!
```
Comment (kneth) [inline examples/electron-todo-list/src/renderer/components/IntroText.tsx:25]: Should WASM be replaced by Electron?

File: examples/electron-todo-list/package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    "build": "craco build",
+    "start": "electron ."
+  },
+  "eslintConfig": {
+    "extends": [
+      "react-app",
+      "react-app/jest"
+    ]
+  },
+  "browserslist": {
```
Comment (kneth) [inline examples/electron-todo-list/package.json:30]: Do we need this section?

**Review by kneth — APPROVED** [review]

File: examples/electron-todo-list/public/electron.cjs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+} catch (_) {}
+
+let mainWindow;
+
+function createWindow() {
+  // Create the browser window.
+  mainWindow = new BrowserWindow({
+    width: 800,
+    height: 600,
+    webPreferences: { nodeIntegration: true, contextIsolation: false },
```
Comment (elle-j) [inline examples/electron-todo-list/public/electron.cjs:18]: Could we add a comment here why we need to disable `contextIsolation`?

File: examples/electron-todo-list/public/electron.cjs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  });
+  // and load the index.html of the app.
+  console.log(__dirname);
```
Comment (elle-j) [inline examples/electron-todo-list/public/electron.cjs:21]: Leftover?
```suggestion
```

File: examples/electron-todo-list/public/manifest.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (elle-j) [inline examples/electron-todo-list/public/manifest.json:3]:
```suggestion
  "short_name": "Electron App",
  "name": "Electron App Using Atlas Device Sync",
```

File: examples/electron-todo-list/src/renderer/atlas-app-services/config.json (hunk 2 above)
Comment (elle-j) [reply examples/electron-todo-list/src/renderer/atlas-app-services/config.json:2]: I agree 👍 (To grab from the README instructions, we could use the placeholder `"YOUR_APP_ID"`.)

File: examples/electron-todo-list/src/renderer/components/AddTaskForm.tsx (hunk 3 above)
Comment (elle-j) [reply examples/electron-todo-list/src/renderer/components/AddTaskForm.tsx:44]: I guess it's a matter of personal preference, but the Wasm example uses these options as well.

File: examples/electron-todo-list/src/renderer/pages/ErrorPage.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+import { isRouteErrorResponse, useNavigate, useRouteError } from 'react-router-dom';
+
+import styles from '../styles/ErrorPage.module.css';
+
+export function ErrorPage() {
+  const navigate = useNavigate();
+  const error = useRouteError();
+
+  console.error(error);
```
Comment (elle-j) [inline examples/electron-todo-list/src/renderer/pages/ErrorPage.tsx:27]:
```suggestion
```

File: examples/electron-todo-list/src/renderer/pages/LoginPage.tsx [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+  };
+
+  return (
+    <div className={styles.container}>
+      <img
+        src={logo}
+        alt='Realm by MongoDB'
+      />
+      <h1>
+        Log in to try out Realm Web & Sync
```
Comment (elle-j) [inline examples/electron-todo-list/src/renderer/pages/LoginPage.tsx:65]:
```suggestion
        Log in to try out Realm in Electron & Atlas Device Sync
```

**Review by elle-j — APPROVED** [review]
Awesome, mainly added suggestions to the README 👍

File: examples/electron-todo-list/src/renderer/atlas-app-services/config.json (hunk 2 above)
Comment (takameyer) [reply examples/electron-todo-list/src/renderer/atlas-app-services/config.json:2]: For sure, I did not intend to commit an actual app-id here

File: examples/electron-todo-list/src/renderer/components/AddTaskForm.tsx (hunk 3 above)
Comment (takameyer) [reply examples/electron-todo-list/src/renderer/components/AddTaskForm.tsx:44]: As the renderer will always be a Chrome browser window, we can safely remove this.

File: examples/electron-todo-list/src/renderer/components/IntroText.tsx (hunk 4 above)
Comment (takameyer) [inline examples/electron-todo-list/src/renderer/components/IntroText.tsx:25]:
```suggestion
        Welcome to a MongoDB Realm, Electron, and Sync app!
```

File: examples/electron-todo-list/src/renderer/components/AddTaskForm.tsx (hunk 3 above)
Comment (takameyer) [inline examples/electron-todo-list/src/renderer/components/AddTaskForm.tsx:44]:
```suggestion
```

File: examples/electron-todo-list/src/renderer/atlas-app-services/config.json (hunk 2 above)
Comment (takameyer) [inline examples/electron-todo-list/src/renderer/atlas-app-services/config.json:2]:
```suggestion
  "ATLAS_APP_ID": "YOUR_APP_ID"
```

File: examples/electron-todo-list/package.json [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+      ">0.2%",
+      "not dead",
+      "not op_mini all"
+    ],
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
```
Comment (takameyer) [inline examples/electron-todo-list/package.json:41]:
```suggestion
```

File: examples/electron-todo-list/package.json [hunk 12]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "realm-electron",
```
Comment (takameyer) [inline examples/electron-todo-list/package.json:2]:
```suggestion
  "name": "@realm/example-electron-todo-list",
```

File: examples/electron-todo-list/public/electron.cjs (hunk 6 above)
Comment (takameyer) [reply examples/electron-todo-list/public/electron.cjs:16]: Done

File: examples/electron-todo-list/README.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+│      ├── components      - React components for the app
+│      ├── hooks           - Custom hooks needed for the app
+│      ├── models          - The local database models
+│      ├── pages           - Routes for the application
+│      ├── styles          - All stylings contained here
+│      ├── App.tsx         - Main app entry point
+│      └── AuthenticatedApp.tsx  - App after auth established
+├── public              - The folder of static contents to the web app
+│   └── electron.js     - The bootstrap script for electron
+└── craco.config.cjs    - The bundler config for the rendering process
```
[the remark this answers is not in the document]
Comment (takameyer) [reply examples/electron-todo-list/README.md]: I've brought those in as well.  Good catch

**Pull request merged**